Beis.LearningPlatform.BL/Services/EmailService.cs

```diff
@@ -224,11 +224,71 @@ async Task<IServiceResponse> IEmailService.SendResultsRemail(Guid requestID, str
 
                 return personalisation;
             }
+            else if (dto is SkilledModuleThreeDto)
+            {
+                var emailData = (SkilledModuleThreeDto)dto;
+
+                templateId = GetSkillsModuleThreeTemplateID(emailData.UserTypeActionPlanSection);
+                
+                personalisation = GeneratePersonalisation(EmailPersonalisationNames.SkillsModuleThree,
+                                                new[] { emailData.QuestionOneStart, emailData.QuestionOneNext, emailData.QuestionOneFinally,
+                                                        emailData.QuestionTwoStart, emailData.QuestionTwoNext, emailData.QuestionTwoFinally,
+                                                        emailData.QuestionThreeStart, emailData.QuestionThreeNext, emailData.QuestionThreeFinally,
+                                                        GetUnsubscribeLink(emailAddress) },
+                                                false);
+
+                return personalisation;
+            }
 
 
             return personalisation;
         }
 
+        private string GetSkillsModuleThreeTemplateID(string userTypeActionPlanSection)
```

Comment:
see if you can wrap the magic strings as a description attribute of the enum and then just convert to respective type and return rather than having big case statements. There are couple of other places like this in the pr to be changed.